pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -88,12 +87,11 @@ func (pl *NodeAffinity) Score(ctx context.Context, state *framework.CycleState,
 			}
 
 			// TODO: Avoid computing it for all nodes if this becomes a performance problem.
```

Comment:
combining the selector construction and match evaluation into a single method makes this TODO more difficult to resolve... in the future, it would be helpful to do signature refactoring separately from a move to component-helpers... changing and moving in the same PR is difficult to review